mantidproject/mantid
# Muon shade region

**Description of work.**



**To test:**


Enable the model analysis
Open muon analysis
Load MUSR 62260
Turn errors on and off. This should look unchanged from the current version (i.e. bars on the data)
Go to the fitting tab
Add a LinearBackground
Do a fit
Turn the errors on and off
This time the data will still have bars, but the fit will have a shaded region instead

Load MUSR 62260-5
Tick simultaneous 
Do a fit
The simultaneous fit, will not have any errors on the fit so instead, use the script below to add some (only do this when errors are off)

```

name = "MUSR62260; Group; top; Asymmetry; MA; Fitted; LinearBackground; Workspace; Simultaneous"
name2 = "MUSR62260; Group; bkwd; Asymmetry; MA; Fitted; LinearBackground; Workspace; Simultaneous"
name3 = "MUSR62260; Group; fwd; Asymmetry; MA; Fitted; LinearBackground; Workspace; Simultaneous"
name4 = "MUSR62260; Group; bottom; Asymmetry; MA; Fitted; LinearBackground; Workspace; Simultaneous"
 
ws0 = mtd[name]

ws = ws0
y= ws.dataE(1)
x = ws.readX(1)
for i in range(len(y)):
    y[i]= 0.1*0.01*x[i]
# 
AnalysisDataService.addOrReplace(name, ws)

ws2 = mtd[name2]

ws2=ConvertToPointData(ws2)
y= ws2.dataE(1)
x = ws2.readX(1)
t =  0.1*np.sin(x)
for i in range(len(y)):
    y[i]=t[i]
AnalysisDataService.addOrReplace(name2, ws2)

ws3 = mtd[name3]

ws3=ConvertToPointData(ws3)
y= ws3.dataE(1)
x = ws3.readX(1)
t= 0.1*np.cos(.5*x)
for i in range(len(y)):
    y[i]=t[i]
AnalysisDataService.addOrReplace(name3, ws3)

ws4 = mtd[name4]

ws4=ConvertToPointData(ws4)
y= ws4.dataE(1)
x = ws4.readX(1)
t= 0.1*np.exp(-x)
for i in range(len(y)):
    y[i]=t[i]
AnalysisDataService.addOrReplace(name4, ws4)
```

Go to the sequential tab
Do fit all
Go to the results tab
Select run number and then create the results table
Go to model analysis
Set x to be run number and y to be a fitting parameter
Add a FlatBackground and do a fit
The fit should have a shaded region when errors are turned on
Go back to the fitting tab
Change the mode from normal to TF asymmetry
Clear the fit function
Add a GausOsc
Set frequency to 1
Do a fit
Check that the fit has a shaded region instead of errors
Close muon analysis


Clear the ADS
Open frequency domain analysis
Load MUSR 62260
Go to the transform tab
Change it from FFT to MaxEnt
Click calculate - it takes a little while
Go to the fitting tab
Add a flat background
Do a fit
When the errors are turned on the fit should have a shaded region. The frequency data will not have any errors.

Fixes #32386. 






---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
